core/src/main/java/org/apache/druid/java/util/common/guava/ParallelMergeCombiningSequence.java

```diff
@@ -191,7 +191,8 @@ public boolean hasNext()
               {
                 final long thisTimeoutNanos = timeoutAtNanos - System.nanoTime();
                 if (hasTimeout && thisTimeoutNanos < 0) {
-                  throw new QueryTimeoutException("Sequence iterator timed out");
+                  throw new QueryTimeoutException("Query did not complete within configured timeout period. " +
+                      "You can increase query timeout or tune the performance of query");
```

Comment:
Suggestion:

> Query did not complete within the timeout period set by druid.some.config.

The action suggested here is OK, but it does not tell the user how to change the config. Also, there could be other causes: network issues, machine is overloaded, too many threads, and on and on.

Reply:
it's tricky. I cannot put all the kinds of nuances in the error message. I intentionally avoided writing the configuration. For one reason, the timeout can be specified in different places. it can come from context, default timeout, and default cluster-wide configuration. 

this is where the error code may help where we return an error code and ask users to look up the documentation for more info on that error code. 

btw, If I were writing this after your PR (error framework) is merged, how will this code be written?

Reply:
As you noted, in another PR, we are trying to improve our error system. This is a perfect example. _If_ we had that system available (which we don't yet), we could solve by throwing, say, a `QueryTimeoutError` with the following message and context:

```text
Query timed out.
Timeout seconds: xx
Suggestion: Review query logs to improve performance, or change the timeout period.
```

It seems that the code above only knows the timeout time, but not the original timeout period. This would be a case where a higher bit of code would catch the timeout exception and fill in the `Timeout seconds` value.

The `Suggestion` applies to simple Druid installations, but perhaps not to managed Druid where an end user has no access to the logs or the config parameters. Such systems could strip out the suggestion or substitute their own.

But, of course, we don't have this system yet. The result is that the above rather lackluster message is about all we can offer at the moment.